src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -653,6 +661,155 @@ MultiClientCopyData(int32 connectionId, int32 fileDescriptor)
 
 
 /*
+ * MultiClientCreateWaitInfo creates a WaitInfo structure, capable of keeping
+ * track of what maxConnections connections are waiting for; to allow
+ * efficiently waiting for all of them at once.
+ *
+ * Connections can be added using MultiClientAddWait(). All added connections
+ * can then be waited upon together using MultiClientWait().
+ */
+WaitInfo *
+MultiClientCreateWaitInfo(int maxConnections)
+{
+	WaitInfo *waitInfo = palloc(sizeof(WaitInfo));
+
+	waitInfo->allocated = maxConnections;
+	waitInfo->waiting = 0;
+	waitInfo->pollfds = palloc(maxConnections * sizeof(struct pollfd));
```

Comment:
Wait, how does setting an array of pointers to zero hide bugs? Wouldn't the first attempt to use one segfault?